### PR TITLE
generate DLL

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: lib/lz4.h [hunk 1]
```diff
@@ -55,12 +55,15 @@ extern "C" {
 *  LZ4_DLL_EXPORT :
 *  Enable exporting of functions when building a Windows DLL
 */
-#if defined(_WIN32) && defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
+#if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
 #  define LZ4LIB_API __declspec(dllexport)
+#elif defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)
+#  define LZ4LIB_API __declspec(dllimport) /* It isn't required but allows to generate better code, saving a function pointer load from the IAT and an indirect jump.*/
```
Comment (Cyan4973) [inline lib/lz4.h]: `dllimport` setting causes troubles with static library compilation. See @#246.
How is it enabled ?

File: lib/lz4frame.h [hunk 2]
```diff
@@ -55,12 +55,10 @@ extern "C" {
...
-#if defined(_WIN32)
-#  if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
-#    define LZ4FLIB_API __declspec(dllexport)
-#  else
-#    define LZ4FLIB_API __declspec(dllimport)
-#  endif
+#if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
+#  define LZ4FLIB_API __declspec(dllexport)
+#elif defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)
+#  define LZ4FLIB_API __declspec(dllimport)
```
Comment (Cyan4973) [inline lib/lz4frame.h]: same question as `lz4.h` regarding `dllimport`

File: lib/lz4hc.h [hunk 3]
```diff
@@ -51,12 +51,10 @@ extern "C" {
...
-#if defined(_WIN32)
-#  if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
-#    define LZ4HCLIB_API __declspec(dllexport)
-#  else
-#    define LZ4HCLIB_API __declspec(dllimport)
-#  endif
+#if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
+#  define LZ4HCLIB_API __declspec(dllexport)
+#elif defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)
+#  define LZ4HCLIB_API __declspec(dllimport)
```
Comment (Cyan4973) [inline lib/lz4hc.h]: and same `dllimport` question here

File: tests/Makefile [hunk 4]
```diff
@@ -85,6 +85,14 @@ fullbench  : $(LZ4DIR)/lz4.o $(LZ4DIR)/lz4hc.o $(LZ4DIR)/lz4frame.o $(LZ4DIR)/xx
 fullbench32: $(LZ4DIR)/lz4.c $(LZ4DIR)/lz4hc.c $(LZ4DIR)/lz4frame.c $(LZ4DIR)/xxhash.c fullbench.c
 	$(CC) -m32 $(FLAGS) $^ -o $@$(EXT)
 
+fullbench-lib: fullbench.c $(LZ4DIR)/xxhash.c
+	$(MAKE) -C $(LZ4DIR) liblz4.a
+	$(CC) $(FLAGS) $^ -o $@$(EXT) $(LZ4DIR)/liblz4.a
+
+fullbench-dll: fullbench.c $(LZ4DIR)/xxhash.c
+	$(MAKE) -C $(LZ4DIR) liblz4
+	$(CC) $(FLAGS) $^ -o $@$(EXT) -DLZ4_DLL_IMPORT=1 $(LZ4DIR)/liblz4.dll.a
```
Comment (inikep) [inline tests/Makefile]: It's used only here.

File: lib/lz4.h (hunk 1 above)
Comment (inikep) [inline lib/lz4.h]: It's a future proof solution. By default it's turned off. One have to define LZ4_DLL_IMPORT=1. So far it's used only with `fullbench-dll`.

**Pull request merged**